src/internal/m365/onedrive/permission.go

```diff
@@ -82,6 +83,56 @@ func getCollectionMetadata(
 	return meta, nil
 }
 
+// Unlike permissions, link shares inherit them from all the parents
+// of an item and not just the direct parent.
+func computePreviousLinkShares(
+	ctx context.Context,
+	originDir path.Path,
+	parentMetas map[string]metadata.Metadata,
+) ([]metadata.LinkShare, error) {
+	var (
+		parent     path.Path
+		linkShares []metadata.LinkShare
+	)
+
```

Comment:
nit: we should add the starting path so we have it
```suggestion
	)

	ctx = clues.Add("item_path", originalDir)
```